beacon-chain/core/helpers/slot_epoch.go

```diff
@@ -148,6 +150,19 @@ func CurrentSlot(genesisTimeSec uint64) uint64 {
 	return uint64(now-genesis) / params.BeaconConfig().SecondsPerSlot
 }
 
+// ValidateSlotClock validates a provided slot against the local
+// clock to ensure slots that are unreasonable are returned with
+// an error.
+func ValidateSlotClock(slot uint64, genesisTimeSec uint64) error {
```

Comment:
I dont think this needs to be exported (yet), it's only usage is within `helper` pkg

Reply:
I take it back. I will use this in my follow up PRs for other pkgs